### PR TITLE
feat(injector): support type injection on `Injector`

### PR DESCRIPTION
**Pull request by TheNightmareX**
### Summary of changes

support type injection on `Injector`

### Relinquishment of Rights

Please mark following checkbox to confirm that you relinquish all rights of your changes:

- [x] I waive and relinquish all rights regarding this changes (including code, text, and images) to Deepkit UG (limited), Germany. This changes (including code, text, and images) are under MIT license without name attribution, copyright notice, and permission notice requirement.

### CODE REVIEW

**Comment by marcj** [comment]
@TheNightmareX can you rebase and solve the conflicts, please?

File: packages/app/src/app.ts [hunk 1]
```diff
@@ -8,20 +8,37 @@
...
+import {
+    EventDispatcher,
+    EventListener,
+    EventListenerCallback,
+    EventOfEventToken,
+    EventToken,
+} from '@deepkit/event';
+import { ReceiveType, ReflectionClass, ReflectionKind } from '@deepkit/type';
+
+export function setPartialConfig(
```
Comment (marcj) [inline packages/app/src/app.ts:37]: unrelated changes

Comment (marcj) [reply packages/app/src/app.ts:37]: actually, I just saw the whole file contains a lot unrelated changes

**Comment by TheNightmareX** [comment]
oh sorry I forgot to force-push to remove the wrong commit

**Comment by marcj** [comment]
Thanks! merged in https://github.com/deepkit/deepkit-framework/commit/8c675ecd26f319c82d9a375fa5ba9edd87e30de1, here again I wrongly pulled it, so it lost author information, sorry!
